Convert Swagger to TypeSpec for databoxedge

haolingdong-msft opened the pull request:
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

<a href="?expand=1&template=sdk_configuration_template.md">Click here</a> to open a PR for only SDK configuration.

haolingdong-msft committed:
migrate once

haolingdong-msft committed:
Fix error type for operations interface

haolingdong-msft committed:
add readonly visibility for paging model value

haolingdong-msft committed:
fix x-ms-secret

haolingdong-msft committed:
update swagger

haolingdong-msft committed:
update swagger

welovej committed:
Update cspell.yaml

welovej committed:
Revert "Update cspell.yaml"

This reverts commit 85cfcc958e2618357db23efba65411c8be30b631.

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
Delete databoxedge.json

welovej committed:
Update readme.md

welovej committed:
prettier

welovej committed:
Create client.tsp

welovej committed:
Update client.tsp

mcgallan committed:
Merge branch 'main' into tsp-conversion-databoxedge

mcgallan committed:
update no properties resource model and summary

mcgallan committed:
update tspconfig

weidongxu-microsoft committed:
Add Java SDK breaking change mitigations for TypeSpec migration

- Added clientName decorators to fix model name casing issues:
  - SSLStatus -> SslStatus
  - NetworkAdapterDHCPStatus -> NetworkAdapterDhcpStatus
  - NetworkAdapterRDMAStatus -> NetworkAdapterRdmaStatus
  - ARMBaseModel -> ArmBaseModel
- Fixed property name changes:
  - RefreshDetails.lastCompletedRefreshJobTimeInUTC -> lastCompletedRefreshJobTimeInUtc
  - StorageAccountCredentialProperties.userName -> username

v-jiaodi committed:
resolve js breaking

ChenxiJiang333 committed:
mitigate reorder breakings (#38379)

weidongxu-microsoft committed:
Update client.tsp

zedy-wj committed:
resolve the conflicts from the folder migration.

mcgallan committed:
Merge branch 'main' into tsp-conversion-databoxedge

mcgallan commented on specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/DataBoxEdge/Addon.tsp:
```diff
@@ -0,0 +1,104 @@
+    SegmentName = "addons",
+    NamePattern = ""
+  >;
+}
```
@v-jiaodi Now that this issue has been resolved, please test whether it will still block SDK validation.

jliusan committed:
add config (#38636)

tadelesh committed:
Add 'go' support to DataBoxEdge client overrides

Updated override annotations to include 'go' alongside 'python' for various methods in the DataBoxEdge client.

mcgallan committed:
format

pshao25 committed:
Merge branch 'main' into tsp-conversion-databoxedge

mcgallan committed:
update

mcgallan committed:
Merge branch 'tsp-conversion-databoxedge' of https://github.com/Azure/azure-rest-api-specs into tsp-conversion-databoxedge
